ash-bergs/pantry-pal
# feat: add uploading to options sidebar

- Adds `syncUpload` method on the `ItemManager` class to add `Items` from JSON file to the list 
- Adds "Upload Backup Data" form to the bottom of the options sidebar and `uploadBackupForm.ts` functionality module 
- Renames the previous generic "item form" and "form" classnames to `addItemForm` for clarity 

<img width="371" alt="image" src="https://github.com/user-attachments/assets/6c219882-655e-4296-9988-c328bcc5d14b">


## Testing 
1. `npm run build` 
2. `npm run dev`
3. Clear site cache and add a few items to the list 
4. Open options
5. Download your list data 
6. Remove one item from the list 
7. Open options modal and click to select the downloaded file from downloads 
8. Upload
9. Items should be added to the list, but not ones that were already in the list (no duplicates)
